src/subscription_manager/repolib.py

```diff
@@ -60,13 +60,98 @@ def manage_repos_enabled():
+        except ValueError as err:
+            log.exception(err)
+            auto_enable_yum_plugins = True
+        except configparser.Error as err:
+            log.exception(err)
+            auto_enable_yum_plugins = True
+        else:
+            if auto_enable_yum_plugins is None:
+                auto_enable_yum_plugins = True
+        return bool(auto_enable_yum_plugins)
+
+    @staticmethod
+    def warning_message(enabled_yum_plugins):
+        message = _('The yum plugins: %s were automatically enabled for the benefit of '
+                    'Red Hat Subscription Management. If not desired, use '
+                    '"subscription-manager config --rhsm.auto_enable_yum_plugins=0" to '
+                    'block this behavior.\n') % str(tuple(enabled_yum_plugins))
+        return message
+
+    @classmethod
+    def enable_yum_plugins(cls):
+        """
+        This function tries to enable yum plugins: subscription-manager and product-id.
+        It takes no action, when automatic enabling of yum plugins is disabled in rhsm.conf.
+        :return: It returns list of enabled plugins
+        """
+
+        # When user doesn't want to automatically enable yum plugins, then return empty list
+        if cls.is_auto_enable_enabled() is False:
+            log.info('The rhsm.auto_enable_yum_plugins is disabled')
+            return []
+
+        log.debug('The rhsm.auto_enable_yum_plugins is enabled')
+
+        # List of successfully enabled plugins
+        enabled_yum_plugins = []
+
+        # Go through the list of yum plugins and try to find configuration
+        # file of these plugins.
+        for yum_plugin_name in cls.YUM_PLUGINS:
+            yum_plugin_file_name = cls.YUM_PLUGIN_DIR + '/' + yum_plugin_name + '.conf'
+            yum_plugin_config = ConfigParser()
+            result = yum_plugin_config.read(yum_plugin_file_name)
+            if len(result) == 0:
+                log.info('Configuration file of yum plugin: "%s" cannot be read' % yum_plugin_file_name)
+                continue
+            is_plugin_enabled = yum_plugin_config.getint('main', 'enabled')
+            if is_plugin_enabled == cls.YUM_PLUGIN_ENABLED:
+                log.debug('Yum plugin: "%s" already enabled. Nothing to do.' % yum_plugin_file_name)
+            elif is_plugin_enabled == cls.YUM_PLUGIN_DISABLED:
+                log.warn('Enabling yum plugin: "%s".' % yum_plugin_file_name)
+                # Change content of plugin configuration file and enable this plugin.
+                with open(yum_plugin_file_name, 'w') as cfg_file:
+                    yum_plugin_config.set('main', 'enabled', cls.YUM_PLUGIN_ENABLED)
+                    yum_plugin_config.write(cfg_file)
+                enabled_yum_plugins.append(yum_plugin_file_name)
+            else:
+                log.warn('Value of rhsm.auto_enable_yum_plugins: "%s" is invalid.' % is_plugin_enabled)
```

Comment:
Is it the rhsm.auto_enabled_yum_plugins value that is invalid? Or is it main.enabled (for yum_plugin_file_name)?

I think this should be changed to `"Invalid value ('%s') found in yum plugin configuration file '%s'" % (yum_plugin_file_name, is_plugin_enabled)`

Or something similar

Reply:
Yeah, you are right. This log message is wrong. There should be this: `"Invalid value ('%s') found in yum plugin configuration file '%s'" % (is_plugin_enabled, yum_plugin_file_name)"`